source/brailleDisplayDrivers/alva.py

```diff
@@ -204,35 +205,48 @@ def terminate(self):
 				# We must sleep after closing the COM port, as it takes some time for the device to disconnect.
 				time.sleep(self.timeout)
 
-	def _ser6SendMessage(self, cmd, value=""):
-		if isinstance(value, (int, bool)):
-			value = chr(value)
-		self._dev.write(b"\x1b{cmd}{value}".format(cmd=cmd, value=value,))
+	def _ser6SendMessage(self, cmd: bytes, value: bytes = b"") -> None:
+		if not isinstance(value, bytes):
+			raise TypeError("Expected param 'value' to have type 'bytes'")
+		self._dev.write(b"".join(["\x1b", cmd, value]))
 
-	def _ser6OnReceive(self, data):
-		if data != b"\x1b": # Escape
+	def _ser6OnReceive(self, data: bytes):
+		"""
+		@type data: bytes
+		"""
+		if data != b"\x1b":  # Escape
 			return
 		cmd = self._dev.read(1)
-		value = self._dev.read(ALVA_SER_CMD_LENGTHS[cmd])
-
-		if cmd == b"K": # Input
-			self._handleInput(ord(value[0]), ord(value[1]))
-		elif cmd == b"E": # Braille cell count
-			self.numCells = ord(value)
-		elif cmd == b"?": # Device ID
-			self._deviceId = ord(value)
-		elif cmd == b"r": # Raw keyboard messages enable/disable
-			self._rawKeyboardInput = bool(ord(value))
-		elif cmd == b"H": # Time
+		value: bytes = self._dev.read(ALVA_SER_CMD_LENGTHS[cmd])
+
+		if cmd == b"K":  # Input
+			self._handleInput(group=value[0], number=value[1])
+		elif cmd == b"E":  # Braille cell count
+			self.numCells = ord(value)  # this command only gets one byte

```

Comment:
To be consistent with the line above, I think value[0] could be considered. May be we should come up with some standard here, i.e. always use ord for single bytes, or use index 0 for single bytes.

Reply:
The pattern I have started to fall into with this is as you said, use ord when you know it is a single byte, and `__getitem__` when it is larger. This case meets that, the comment is not particularly insightful, at this point I was hoping to be able to confirm the sizes of all the commands (`cmd` values). But it is too time consuming in the scope of this PR.